### PR TITLE
feat: Add support for C#

### PR DESCRIPTION
**Pull request by NobleMathews**
Implements task from https://github.com/deepsourcelabs/murdock/issues/21

Adds support for C#

Checks .nuspec file for package metadata

Retrieves data from NuGet API

### CODE REVIEW

File: dependencies/go/go_worker.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def handle_go_mod(req_file_data: str) -> Result:
     free(ptr)
     d = json.loads(out)
     m = {
-        "MinGoVer": "lang_ver",
+        "MinGoVer": "lang_ve    r",
```
Comment (pawan-deepsource) [inline dependencies/go/go_worker.py:50]: Is this intentional?

Comment (NobleMathews) [reply dependencies/go/go_worker.py]: Nope my bad, fixed thanks!

File: tests/test_inspector.py [hunk 2]
```diff
@@ -161,3 +161,23 @@ def test_unsupported_vcs_fails(result_payload):
     """Checks if exception is raised for unsupported pattern"""
     with pytest.raises(VCSNotSupportedError, match="gitlab"):
         inspector.handle_vcs("go", "gitlab.com/secmask/awserver", result_payload)
+
+
+def test_make_single_request_cs(psql):
+    """Test version and license for javascript"""
```
Comment (tushar-deepsource) [inline tests/test_inspector.py:167]: s/javascript/c#

File: tests/test_inspector.py [hunk 3]
```diff
@@ -161,3 +161,23 @@ def test_unsupported_vcs_fails(result_payload):
...
+    result = inspector.make_single_request(
+        psql, "cs", "Microsoft.Bcl.AsyncInterfaces", force_schema=False
+    )[0]
+    assert result
+
+
+def test_make_single_request_cs_ver(psql):
+    """Test version and license for javascript"""
```
Comment (tushar-deepsource) [inline tests/test_inspector.py:175]: s/javascript/c#

**Review by tushar-deepsource — CHANGES_REQUESTED** [review]

**Review by tushar-deepsource — APPROVED** [review]

**Pull request merged**